### DISCUSSION TITLE
[Art competition] Program Synthesis

### DISCUSSION THREAD

**Discussion by done-4u** [prosyslab-classroom/cs424-program-reasoning#251] [Art competition]
![image](https://user-images.githubusercontent.com/44857899/204651471-0a321380-9727-434d-8f40-da60b56f1d91.png)

## Short Description
For me, [*program synthesis*] means [*picking flowers on the cliff*].

## Long Description (up to 3 sentences)
It is not easy to synthesize the desired program from a large search space, like picking flowers on the cliff. They both impress my heart when I finally achieve the goal.

**Comment by KihongHeo** [comment]
Good. Nice try.

Can we have a more closely related analogy? Any non-trivial challenge can be described like this. What would be a unique point of program synthesis?
